src/services/contactReport.service.js

```diff
@@ -0,0 +1,24 @@
+import { $db } from '@/services/firebase.service'
+class ContactReportService {
+  async getContactsOfUserByID (userId, date) {
+    const userList = []
+    const userRef = await $db().doc('User/' + userId)
+    const querySnapshot = await $db().collectionGroup('CheckIn')
+      .where('date', '==', date)
+      .where('user', 'array-contains', userRef)
+      .get()
+
+    querySnapshot.forEach((doc) => {
+      const userRefs = doc.data().user
+      userRefs.forEach(u => {
+        u.get().then(user => {
+          userList.push(user.data())
+        })
+      })
+    })
```

Comment:
```suggestion
    querySnapshot.forEach((doc) => {
      const userRefs = doc.data().user
      userRefs.forEach(async (userRef) => {
        const user = await userRef.get()
        userList.push(user.data())
      })
    })
```

Reply:
@cedi315 Why do you suggest to slow down by breaking async calls using await?